### ISSUE TITLE
Resource Datastore Tab Upload Log Messages are not sorted properly in sequential order 

### ISSUE THREAD

**Issue by jqnatividad** [ckan/ckan#7342]
## CKAN version
2.9.7

## Describe the bug
The [`ckanext/datapusher/templates/datapusher/resource_data.html`](https://github.com/ckan/ckan/blob/ckan-2.9.7/ckanext/datapusher/templates/datapusher/resource_data.html) does have logic to sort log entries.
However, the resolution is only down to the second. With Datapusher and XLoader, this is not a problem as their log messages don't have subsecond resolution.

However, with Datapusher+ verbose logging, there might be multiple log messages with the same timestamp to the second. 

### Steps to reproduce
Push a file to the Datastore using Datapusher+ and reload DataStore tab. Often, the log messages are listed correctly, but sometimes, they are out of order as messages with the same timestamp (to the second) are listed randomly next to each other.

### Expected behavior
Upload Log messages are always listed in timestamp order to millisecond precision.

**Comment by EricSoroos** [comment]
One question would be why we're sorting in a template rather than a database, which will likely break with pagination.

**Comment by jqnatividad** [comment]
You're correct @EricSoroos , sorting should be done on the database. I'll take a crack at this.
